project-koku/koku
# [COST-3728] Remove rows when distribute overhead disabled

## Jira Ticket

[COST-3728](https://issues.redhat.com/browse/COST-3728)

## Description

This change will remove the `platform_distributed` & `worker_distributed`  rows, when distribute is false. 

## Testing

Check database:
```
select distinct(cost_model_rate_type) from reporting_ocpusagelineitem_daily_summary WHERE cluster_id='my-ocp-cluster-3' AND (cost_model_rate_type = 'platform_distributed' OR cost_model_rate_type = 'worker_distributed') AND usage_start >= '2023-04-01';
```

1. Checkout Branch
2. make load-test-customer data.
3. Edit the openshift on prem cost model:
**Remove only one:**
edit cost model
```
"distribution_info": {
        "distribution_type": "cpu",
        "platform_cost": false,
        "worker_cost": true
    }
```
Check database Result:
```
 cost_model_rate_type
----------------------
 worker_distributed
(1 row)
```

*Add both*
edit cost model
```
"distribution_info": {
        "distribution_type": "cpu",
        "platform_cost": true,
        "worker_cost": true
    }
```
Check database Result:
```
 cost_model_rate_type
----------------------
 worker_distributed
 platform_distributed
```
*Remove Both*
edit cost model
```
"distribution_info": {
        "distribution_type": "cpu",
        "platform_cost": false,
        "worker_cost": false
    }
```
check database result:
```
 cost_model_rate_type
----------------------
(0 rows)
```

## Notes

...
